modules/internal/ArrayViewRankChange.chpl

```diff
@@ -402,7 +402,9 @@ module ArrayViewRankChange {
     override proc dsiAlignedLow return upDom.dsiAlignedLow;
     override proc dsiAlignedHigh return upDom.dsiAlignedHigh;
     override proc dsiIndexOrder(i) return upDom.dsiIndexOrder(i);
-    override proc dsiMakeIndexBuffer(size) return upDom.dsiMakeIndexBuffer(size);
+    deprecated "dsiMakeIndexBuffer has been renamed to dsiCreateIndexBuffer"
+    override proc dsiMakeIndexBuffer(size) return dsiCreateIndexBuffer(size);
```

Comment:
Should this be `... return upDom.dsiCreateIndexBuffer(size)`?

Also, are these "dsi" functions purely internal? If so, I don't think it's necessary to mark them as deprecated to change the names.

Reply:
I think it's the same either way since it calls `dsiCreateIndexBuffer` which in turn calls `upDom.dsiCreateIndexBuffer`. I made the old name call the new named one as opposed to  just copying its body, to avoid code duplication.

As for the `dsi` functions being internal, I think you're right; they don't show up in a docs search. I'll change those without deprecation.